hwaccess: sys/io.h is not specific to glibc

Under Linux, sys/io.h provides inb and outb, so we really need it.
However, its inclusion is conditional to the _GLIBC_ define. This is
usually OK under Linux, since both glibc and uClibc define it (uclibc
fakes being glibc).

But the musl C library does not impersonate glibc, so we're missing
including sys/io.h in this case.

Change the include from checking _GLIBC_ to checking whether this is
Linux, looking for the __linux__ define.

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>

diff -durN flashrom-0.9.8.orig/hwaccess.h flashrom-0.9.8/hwaccess.h
--- flashrom-0.9.8.orig/hwaccess.h	2015-02-10 09:03:10.000000000 +0100
+++ flashrom-0.9.8/hwaccess.h	2015-10-28 20:01:54.259202484 +0100
@@ -27,7 +27,7 @@
 #include "platform.h"
 
 #if IS_X86
-#if defined(__GLIBC__)
+#if defined(__linux__)
 #include <sys/io.h>
 #endif
 #endif
